events_test.go

```diff
@@ -32,58 +22,39 @@ func TestCloseEventStream(t *testing.T) {
 
 func TestCloseEventSubscriber(t *testing.T) {
 	es := NewEventSubscriber()
-	errc := make(chan error)
-	go func() {
-		errc <- es.Close()
-	}()
-	select {
-	case err := <-errc:
-		if err != nil {
-			t.Errorf("closing returned error: %v", err)
-		}
-	case <-time.After(3 * time.Second):
-		t.Errorf("timeout!")
+	if err := es.Close(); err != nil {
+		t.Errorf("closing returned error: %v", err)
 	}
+
 }
 
 func TestReceiveSpecificKeyEvent(t *testing.T) {
-	done := make(chan bool)
-	go func() {
-		es := NewEventSubscriber()
-		t.Log("Subscribe")
-		evtChan := es.Subscribe(sdl.K_UP)
-		t.Log("Create Fake Event")
-		fakeEvent := &sdl.KeyDownEvent{
-			Keysym: sdl.Keysym{
-				Sym:      sdl.K_UP,
-				Scancode: sdl.SCANCODE_UP,
-				Mod:      sdl.KMOD_NONE,
-				Unicode:  0,
-			},
-			Type:      sdl.KEYDOWN,
-			Timestamp: 0,
-			WindowID:  0,
-			State:     sdl.PRESSED,
-			Repeat:    0,
-		}
-		t.Log("Push fake Event")
-		sdl.PushEvent(fakeEvent)
-		select {
-		case evt := <-evtChan:
-			switch e := evt.(type) {
-			case *sdl.KeyDownEvent:
-				if e.Keysym.Sym != sdl.K_UP {
-					t.Errorf("Key is %v instead of %v", e.Keysym.Sym, sdl.K_UP)
-				}
+	es := NewEventSubscriber()
+	evtChan := es.Subscribe(sdl.K_UP)
+	fakeEvent := &sdl.KeyDownEvent{
+		Keysym: sdl.Keysym{
+			Sym:      sdl.K_UP,
+			Scancode: sdl.SCANCODE_UP,
+			Mod:      sdl.KMOD_NONE,
+			Unicode:  0,
+		},
+		Type:      sdl.KEYDOWN,
+		Timestamp: 0,
+		WindowID:  0,
+		State:     sdl.PRESSED,
+		Repeat:    0,
+	}
+	sdl.PushEvent(fakeEvent)
+	select {
+	case evt := <-evtChan:
+		switch e := evt.(type) {
+		case *sdl.KeyDownEvent:
+			if e.Keysym.Sym != sdl.K_UP {
+				t.Errorf("Key is %v instead of %v", e.Keysym.Sym, sdl.K_UP)
 			}
-		case <-time.After(5 * time.Second):
-			t.Error("timeout after 5 seconds")
 		}
-		done <- true
-	}()
-	select {
-	case <-done:
-	case <-time.After(10 * time.Second):
-		t.Error("Timeout Nothing went well")
+	// is this necessary?
+	case <-time.After(1 * time.Millisecond):
```

Comment:
I would remove this and push the functionality in the event handler.
Rationale: The upper limit on response time is either needed due to soft realtime requirements and should be in the API. Or the upper limit is just there for sanity and can be done with with `go test -timeout=1s`